git-ignore.sh

```diff
@@ -77,8 +77,9 @@ get_git_ignore () {
 
 	if test $_parent_level -ne 0
 	then
-	    for i in $(seq 1 $_parent_level)
```

Comment:
If the problem is a missing operand, isn't it more likely that `$_parent_level` is set to empty?

Reply:
interesting... i don't have a windows box to test on, so I was guessing that the seq binary worked differently on windows (which I've seen in other utilities before). I'm not sure how $_parent_level could be empty since it's defaulted on line 162. The only guess I have is on line 182, 
```
_parent_level="${1#--parent-level=}"
```
evaluates to empty on windows... maybe... Tricky to debug without being able to reproduce it. I'll see what I can come up with.

Thanks!

Reply:
One simple trick I use all the time is to insert something like

```sh
test -z "$DDDEBUG" || set -x
```

into the shell script in question, and then run the command in the test suite prefixed with `DDDEBUG=1`, like so:

```sh
DDDEBUG=1 git ignore ...
```

Another trick I use when debugging things via Azure Pipelines (I don't have a macOS setup, for example) is to edit the `azure-pipelines.yml` definition to execute only the job that I am *really* interested in, and also substitute the `make test` in `ci/run-build-and-test.sh` to something like `(cd t && sh t7070-*.sh -i -v -x)`, to accelerate the turnaround between force-push and test result.

Maybe that helps you, too?

Reply:
Awesome tricks which I will be sure to employ. Do you think it's necessary at this point since the last CI succeeded? Also, are there any more suggestions on getting this submitted? I changed the description, but I don't know if that is good enough. I think the most descriptive message I could write about this change is in the original commit message.

Reply:
> Do you think it's necessary at this point since the last CI succeeded?

I fear that the value of `_parent_level` might still be bogus and that the most recent changes might still just paper over that issue...

> are there any more suggestions on getting this submitted?

Yes. The commit https://github.com/gitgitgadget/git/pull/264/commits/4462b7f49a04c48bab03e6a722100c9b2ade5d90 should be squashed into the first commit: On the Git mailing list, adding a patch with a bug that is immediately fixed by the next patch is frowned upon. Better to not even introduce the bug in the first place.

But then, I think that `seq` might be the better choice, still, you just need to figure out why it gets an incorrect number of parameters.

Reply:
line 3659 shows _parent_level is getting set correctly to 2:
```
+ _parent_level=2
```

and line 3691 shows seq is being used correctly:
```
++ seq 1 2
/usr/bin/seq: missing operand
Try '/usr/bin/seq --help' for more information.
```

it's just that version of windows (or mingw) isn't supporting it correctly. I'm not up for troubleshooting why 'seq 1 2' doesn't work on that environment of windows. That's why I went with the while loop to replace it... it seems like a reasonable, less fragile, and more portable implementation than 'seq'.

Thoughts?

Reply:
Hmm. Works here:

```sh
$ /usr/bin/seq 1 2
1
2
```